---
title: "Characterization of the microbial and fungal communities in ancient pottery from Neolithic southwestern China"
author:
  - Li-Ying Wang:
      correspondence: "yes"
      email: liying15@gmail.com
      orcid: 0000-0001-9060-2125
      institute:
        - AS
  - Kuei-Chen Lin:
      institute: AS
      orcid: 0000-0003-1689-0558
  - Zhiqing Zhou:
      orcid: 0000-0003-1689-0551
      institute:
        - name: Chengdu Institute of Cultural Relics and Archaeology
  - Yin-Ru Chiang:
      correspondence: "yes"
      institute: AS
      orcid: 0000-0003-1689-0559
institute:
  - AS:
      name: Academia sinica
      address: 130, Section 2, Academia Rd, Nangang District, Taipei City, Taiwan
title-block-published: "Last updated"  
date: now
date-format: long
format: 
  docx:
    reference-doc: "../templates/template.docx" # Insert path for the DOCX file
execute:
  echo: true
  warning: false
  message: false
  comment: "#>"
  fig-path: "../figures/"
  fig-dpi: 600
filters:
  - ../templates/scholarly-metadata.lua
  - ../templates/author-info-blocks.lua
  - ../templates/pagebreak.lua
bibliography: references.bib
csl: "../templates/journal-of-archaeological-science.csl" # Insert path for the bib-style
abstract: |
  Text of abstract
keywords: |
  keyword 1; keyword 2; keyword 3
highlights: |
  These are the highlights. 
---

<!-- This is the format for text comments that will be ignored during renderings. Do not put R code in these comments because it will not be ignored. -->

<!-- With the following code you can access and display values from the yml header above. -->

Keywords: `r rmarkdown::metadata$keywords`

Highlights: `r rmarkdown::metadata$highlights`

<!-- The actual document text starts here: -->

```{r setting}
#| echo: false
# Format integers
options(scipen=999)
knit_print.integer = function(x, ...) {
  prettyNum(x, big.mark=",")
}

registerS3method(
  "knit_print", "integer", knit_print.integer,
  envir = asNamespace("knitr")
)
```

# Introduction

Ceramics are the most common artifacts unearthed from archaeological sites, offering valuable insights into diverse anthropological facets encompassing cultural, social, and economic dimensions. Foodway is one of the topics. Food residues absorbed into the matrices of ceramics are useful sources to explore pottery use and past culinary practices. This analytical pursuit heavily relies on contemporary scientific methodologies. For example, organic residues, particularly lipids, extracted from pottery through a series of chemical techniques enable the identification of terrestrial animals, marine resources, and plants [@Evershed2008; @Evershed2002; @Evershed2008marine; @Craig2011; @Heron2016]. Furthermore, the identification of foodstuffs contained in storage vessels can be achieved through ancient DNA analysis, providing direct evidence of pottery content at the genes or species levels [@Foley2012; @Hansson2008; @Robinson2017]. These analytical approaches underscore the significance of preserving the condition of food residues within archaeological contexts, ensuring the efficacy of these scientific methodologies.

Recently, the application of next-generation sequencing in microbiology has brought a novel biomolecular approach to examine ancient human activity through microorganisms present in archaeological samples, including human remains [@Cano2000; @Ubaldi1998; @Ckimsis2023], teeth [@Adler2013; @Warinner2015; @De2013; @Preus2011], archaeological deposits, and minerals [@Margesin2017; @Siles2018; @Lenehan2017; @Boivin2023]. Such microbial archaeology could also be useful for studying food residues adhering to or absorbed into pottery matrix. Based on the assumption that food spoilage mainly results from microbial activity, the microorganisms preserved in pottery within burial contexts could be related to the last prehistoric use of the pot, such as food that was cooked in it. For example, microorganisms including *Proteus spp.*, *Weissella spp.*, *Pseudomonas spp.*, and *Enterobacteriaceae* are frequently associated with meat [@Tassew2010; @Stellato2016], while *Aspergillus spp.*, *Penicillium spp.*, *Fusarium spp.*, *Rhizopus spp.* are commonly found in association with rice [@Li2022; @Bertuzzi2019; @Tahir2012]. Moreover, *Acetobacter*, *Lactobacillus*, *Bacillus*, *Bifidobacterium* are indicative of fermented foods [@Tamang2016; @Xiao2020; @Zhang2018], and *Paenibacillus*, *Chryseobacterium* are largely found in mixed rice-vegetable meals [@Helmond2017; @Nde2021]. Given the potential widespread occurrence of microbial communities across various food categories, more studies are still needed to understand the characteristics of microbial communities in ancient pottery. 

This study serves as a pilot study to investigate the feasibility of using microbial communities as indicators of past food processing in pottery by focusing on the potsherds from the Guijiabao site in southwest China. As a long-term occupation site from Neolithic period to historical period, Guijiabao revealed abundant archaeological assemblages and remains to explore the adaptation of farming communities in a mountain setting. We aim to explore the differences in overall microbial communities, including bacterial and fungal communities, through a comparative analysis of different parts of ceramic samples and soil samples. Our main research question is: Can microbial compositions serve as evidence for distinguishing various food categories based on the prevalence of spoilage microorganisms? We hypothesized that the microbial composition in absorbed residues within the pottery matrix will exhibit distinctions relevant to food residues when compared to those on the surface of pottery and in surrounding soils. We also compared the resolution of V1, V4, and V6 16S rRNA regions to estimate microbial community diversity. While microorganisms offer indirect evidence of past food practices, our focus is on their potential to provide valuable information that complements existing scientific approaches to analyzing food residues. This innovative perspective could shed light on food practices from a microbial perspective that has not been fully discussed in archaeology. 

# Site Description and Sampling

Guijiabao is located in the Yanyuan Basin in the middle Hengduan Mountains of southwest China. The chronological framework of Guijiabao includes the Neolithic period (5000-3700 BP), the Bronze Age (3200-2700 BP), and the historical period spanning from the late North Song to early Ming dynasties (1200-500 BP) based on radiocarbon dates [@Hao2022; @Huan2022]. Given its location along the route of population movements, Guijiabao has been a place for cultural exchanges with neighboring regions, including the northern and southern parts of the Hengduan Mountains, the Sichuan Basin, and western Yunnan since the Neolithic period. This culture interactions is evident in the similarities observed in pottery styles and decorations [@Guijiabao2017a]. Excavation at Guijiabao revealed various pottery forms, including pots, jars, urns, bowls, and basins. Predominantly, the pottery types are characterized by grayish high-tempered fine ware and reddish-brown coarse ware featuring cord-marked or basket-impressed patterns [@Hao2022; @Guijiabao2017a]. 

As the earliest Neolithic site in the Yanyuan Basin, Guijiabao provides valuable data for exploring the lives of people who have lived in a highland environment (approximately 2400 m above sea level). Archaeobotanical evidence and phytolith analyses suggested that mixed crop agriculture, specifically rice and millet, was introduced to this region since the occupation of the site [@Huan2022; @Guijiabao2016]. Cultivated crops, such as broomcorn millet, foxtail millet, and rice were identified as the primary agricultural practices at the site [@Deng2019; @Guijiabao2016]. Hunting and fishing could be also part of local subsistence activity, according to the finding of stone hunting implements and net sinkers [@Hao2022; @Guijiabao2020a]. Examining pottery used in daily life becomes critical in shedding light on food practices, offering more evidence for understanding local subsistence via the lens of food preparation.

As a preliminary investigation of the feasibility of using a microbial approach for detecting evidence of cooking and food types, we sampled five potsherds of typical ceramics from Guijiabao, with a pottery sample from Gaoshan, anorther Neolithic site in southwest China, as a comparison. Gaoshan is located in southwest Chengdu Plain, an alluvial plain in Sichuan, and dates back to the early Baodun period, the earliest Neolithic culture (ca. 4700-3700 BP) in the Chengdu Plain [@Gaoshan2017]. Archaeological evidence from the Gaoshan site suggests a primary subsistence pattern involving rice cultivation and pig husbandry, supplemented by millet cultivation [@Gaoshan2017; @JiangGaoshan2017; @Lee2020; @He2020]. In addition to the pottery samples, we collected three soil samples from cultural layers across various excavation units at Guijiabao, alongside two soil samples from Gaoshan.
After sampling, the potsherds were securely stored in sealed boxes, while soil samples were preserved in sterile vials, both of which were subsequently frozen at -80 Â°C in preparation for DNA analyses.

# Methods

To minimize potential sources of contamination from the environments, we cleaned all lab benches, equipment, and pipettors with 3% bleach, followed by 99% ethanol and UV light. The sample preparation and DNA extraction were conducted in the biomolecular laboratory facilities at Academia Sinica, Taiwan. Because modern contamination could be a significant concern when interpreting the data, we used negative controls during sample preparation, and reagent blanks for DNA extraction, and PCR amplification to monitor extraneous DNA from surroundings, reagents, and consumables. A total of 12 powdered pottery samples (interior and exterior parts), 5 soil samples from Guijiabao and Gaoshan, and 3 negative control samples were analyzed in this study.

## Sample preparation and DNA extracation

Each pottery sample was divided into two portions: the interior, representing the middle part of the potsherd matrix, and the exterior surface exposed to the environment. This division aimed to facilitate a comparative analysis of the microbial communities in residues absorbed into the potsherds versus those on the surface. Approximately 1.5 g of each sample was collected using a sterile drill operated at a low speed to prevent heat generation. The powdered sample was initially gathered on pre-combustion foil and immediately transferred to a sterile Eppendorf tube. All the tools were cleaned between each iteration to minimize cross-contamination. For the soil samples, around 1g was collected using sterile spatulas and transferred to a sterile Eppendorf tube. 

Total DNA from each sample was extracted in a portion of 300 mg using the DNeasy PowerSoil Pro kit (Qiagen, Germany), and the DNA extracts from the same sample were pooled. The concentration of DNA was measured using a Qubit 2.0 fluorometer and standardized to 1 ng Î¼L^â1^.

## PCR analyses

Given the fragmentation of DNA in archaeological contexts, we targeted single hypervariable regions, V1, V4, and V6 of 16S rRNA, and ITS1 of 18S rRNA, to cover both bacterial and fungal communities. The V1 region was amplified using the primers 8F (5'-AGAGTTTGATYMTGGCTCAG-3') and 120R (5'-TTACTCACCCGTICGCCRCT-3') [@Adler2013], the V4 region was amplified using arch519F (5â²-CAGCMGCCGCGGTAA-3â²) and arch806R (5â²-GGACTACVSGGGTATCTAAT-3â²), and the V6 region was amplified using 926F (5â-AAACTYAAAKGAATTGACGGG-3â) and 1046R (5â-CGACARCCATGCASCACCT-3â) [@Adler2013; @Siles2018]. For fungal communities, the region ITS1 of 18S rRNA was amplified using the primers ITS1F (5â-CTTGGTCATTTAGAGGAAGTAA-3â) and ITS2 (5â²-GCTGCGTTCTTCATCGATGC-3â²) [@Siles2018]. Barcoded forward adapters were added to the PCR primers to facilitate sample pooling for library preparations. The DNA amplification for bacteria was conducted using the Taq DNA polymerase Master Mix RED (Ampliqon) under the following program: initial denaturation at 95 Â°C for 1 min, followed by 35 cycles of denaturation at 95 Â°C for 30 s, primer annealing at 50 Â°C for 30 s, extension at 72 Â°C for 30 s, and final elongation at 72 Â°C for 5 min. For fungi, the thermal conditions are the same as for bacteria, with the exception of 40 cycles and primer annealing temperature at 57 Â°C for 40 s. 

The DNA fragments in PCR products were purified with the GenepHlow gel extraction kit  (Geneaid Biotech Ltd.) after conducting 2 % agarose gel electrophoresis. The amplified products were then pooled and used to build a DNA library using the Illumina TruSeq DNA library preparation protocol.

## DNA library construction and sequencting

## Taxonomic classification

# Results

```{r get-data}
#| label: get-data
#| echo: false
#| cache: true
library(tidyverse)
library(readxl)
# read original count data
OTU_V4 <- read.table(here::here("analysis","data","raw_data","V4_asv_table.txt"), header = T, sep="\t")
OTU_V1 <- read.table(here::here("analysis","data","raw_data","V1_asv_table.txt"), header = T, sep="\t")
OTU_V6 <- read.table(here::here("analysis","data","raw_data","V6_asv_table.txt"), header = T, sep="\t")
OTU_ITS1 <- read.table(here::here("analysis","data","raw_data","ITS1_asv_table.txt"), header = T, sep="\t")

OTU_reads_total <- function(df) {
  sum_read <- df %>%
    rowwise() %>% 
    mutate(sum_read = sum(across(where(is.numeric)))) %>% 
    select(sum_read)
  return(sum(sum_read))
}

V4_total_reads <- OTU_reads_total(OTU_V4)
V1_total_reads <- OTU_reads_total(OTU_V1)
V6_total_reads <- OTU_reads_total(OTU_V6)
ITS1_total_reads <- OTU_reads_total(OTU_ITS1)
total_reads <- V4_total_reads + V1_total_reads + V6_total_reads + ITS1_total_reads

# read rarefied data
rarefy_V4 <- read.csv(here::here("analysis","data","derived_data", "V4_shared_format.1.subsample.60216.csv"), sep = ",")
rarefy_V1 <- read.csv(here::here("analysis","data","derived_data","V1_shared_format.1.subsample.13658.csv"), sep = "")
rarefy_V6 <- read.csv(here::here("analysis","data","derived_data", "V6_shared_format.1.subsample.28402.csv"), sep = ",")
rarefy_ITS1 <- read.csv(here::here("analysis","data","derived_data", "ITS1_shared_format.1.subsample.387.csv"), sep = ",")
context <- read_excel(here::here("analysis","data", "raw_data", "GJB_amp_meta.xlsx"))

rarefy_reads <- function(df) {
  sum_read <- df %>%
    rowwise() %>% 
    mutate(sum_read = sum(across(starts_with("ASV")))) %>% 
    select(sum_read)
  return(sum(sum_read))
}

V4_rare_reads <- rarefy_reads(rarefy_V4)
V1_rare_reads <- rarefy_reads(rarefy_V1)
V6_rare_reads <- rarefy_reads(rarefy_V6)
ITS1_rare_reads <- rarefy_reads(rarefy_ITS1)
rare_reads <- V4_rare_reads + V1_rare_reads + V6_rare_reads + ITS1_rare_reads

V4_OTU_num <- unique(rarefy_V4$numOtus)
V1_OTU_num <- unique(rarefy_V1$numOtus)
V6_OTU_num <- unique(rarefy_V6$numOtus)
ITS1_OTU_num <- unique(rarefy_ITS1$numOtus)
total_OTU <- V4_OTU_num + V1_OTU_num + V6_OTU_num + ITS1_OTU_num
```

```{r alpha}
#| label: alpha
#| echo: false
#| cache: true
library(vegan) 
# to get the same order for all
rarefy_V4 <- rarefy_V4 %>% mutate(Group = tolower(Group)) %>% arrange(Group)
rarefy_V1 <- rarefy_V1 %>% mutate(Group = tolower(Group)) %>% arrange(Group)
rarefy_V6 <- rarefy_V6 %>% mutate(Group = tolower(Group)) %>% arrange(Group)
rarefy_ITS1 <- rarefy_ITS1 %>% mutate(Group = tolower(Group)) %>% arrange(Group)

# tidy meta data and get the same order as the rarefy data
context_tidy <-
  context %>%
  mutate(`adpater no` = tolower(`adpater no`),
         type = tolower(str_extract(`adpater no`, "[[:alpha:]]+(?=\\d)")),
         type = case_when(type == "con" ~ "control", TRUE ~ as.character(type)),
         area = case_when(area == "inside" ~ "interior", area == "outside" ~ "exterior"),
         group = if_else(!is.na(area)&!area%in% c("belly", "outside belly"),
                         paste0(type, "-", area), type),
         group = factor(group, levels = c("control","pot-interior","pot-exterior","soil"))) %>%
  arrange(`adpater no`)

# function to have OTU reads only
reads_only <- function(df) {
  df %>%
    select(-label, -Group, -numOtus)
}

rarefy_V4_reads_only  <- reads_only(rarefy_V4)
rarefy_V1_reads_only  <- reads_only(rarefy_V1)
rarefy_V6_reads_only  <- reads_only(rarefy_V6)
rarefy_ITS1_reads_only  <- reads_only(rarefy_ITS1)

# function to get context for each region
meta_region <- function(df, region){
  context_tidy %>%
    filter(Region == region) %>%
    arrange(`adpater no`)
}

V4_meta<- meta_region(context_tidy, "V4")
V1_meta<- meta_region(context_tidy, "V1")
V6_meta<- meta_region(context_tidy, "V6")
ITS1_meta<- meta_region(context_tidy, "ITS1")

# functions calculate richness, Chao1, evenness index, and Shannon index 
alpha_cal_df <- function(df1, df2){
  richness <- estimateR(df1)
  evenness <- diversity(df1)/log(specnumber(df1))
  shannon <- diversity(df1, index = "shannon")
  alphadiv <- cbind(df2, t(richness), shannon, evenness) #combine all in one
  return(alphadiv)
}

V4_alphadiv <- alpha_cal_df(rarefy_V4_reads_only, V4_meta)
V1_alphadiv <- alpha_cal_df(rarefy_V1_reads_only, V1_meta)
V6_alphadiv <- alpha_cal_df(rarefy_V6_reads_only, V6_meta)
ITS1_alphadiv <- alpha_cal_df(rarefy_ITS1_reads_only, ITS1_meta)

# pairwise Wilcoxon rank-sum  on richness and shannon
library(rstatix)
rich_pair_V4 <- V4_alphadiv %>% pairwise_wilcox_test(S.obs ~ group, p.adjust.method = 'fdr')
rich_pair_V1 <- V1_alphadiv %>% pairwise_wilcox_test(S.obs ~ group, p.adjust.method = 'fdr')
rich_pair_V6 <- V6_alphadiv %>% pairwise_wilcox_test(S.obs ~ group, p.adjust.method = 'fdr')
rich_pair_ITS1 <- ITS1_alphadiv %>% pairwise_wilcox_test(S.obs ~ group, p.adjust.method = 'fdr')

shan_pair_V4 <- V4_alphadiv %>% pairwise_wilcox_test(shannon ~ group, p.adjust.method = 'fdr')
shan_pair_V1 <- V1_alphadiv %>% pairwise_wilcox_test(shannon ~ group, p.adjust.method = 'fdr')
shan_pair_V6 <- V6_alphadiv %>% pairwise_wilcox_test(shannon ~ group, p.adjust.method = 'fdr')
shan_pair_ITS1 <- ITS1_alphadiv %>% pairwise_wilcox_test(shannon ~ group, p.adjust.method = 'fdr')
rich_pv_v4 <- format.pval(pv = rich_pair_V4$p.adj[6], 
              digits = 2,  # number of digits, but after the 0.0
              eps = 0.05, # above the number will be replaced  by "<0.0xxx"
              nsmall = 2) # how much tails 0 to keep 

rich_pv_v6 <- format.pval(pv = rich_pair_V6$p.adj[2], digits = 2, eps = 0.05, nsmall = 2) 
shan_pv_v4 <- format.pval(pv = shan_pair_V4$p[5], digits = 2, eps = 0.05, nsmall = 2) 
```

## Alpha diversity analysis of bacterial and fungal communities
Illumina analysis generated a total of `r total_reads` sequences that were normalized to `r rare_reads` after rarefying for comparison across different types of samples [@Schloss2024; @Weiss2017]. Soil samples show a significant higher richness index based on the number of observed OTUs than that of pottery exterior samples and pottery interior samples (p `r rich_pv_v4`) for the the V4 and V6 regions (Table S1). Also, the richness index of pottery exterior samples is significant higher than that of control samples for the V6 region (p `r rich_pv_v6`). The richness index did not differ significantly between samples for the V1 and ITS1 regions. Shannon diversity index shows that soil samples are significantly higher than pottery interior samples (p `r shan_pv_v4`; Table S2). However, there is no statistically significant differences in richness index and Shannon diversity index between the two types of pottery samples and negative control. 

```{r taxa-abundance}
#| label: taxa-abundance
#| echo: false
#| cache: true
V4_taxa_ab <- read.csv(here::here("analysis","data","derived_data","V4_bac_phyla_rel_ab.csv"))
V6_taxa_ab <- read.csv(here::here("analysis","data","derived_data","V6_bac_phyla_rel_ab.csv"))
ITS1_taxa_ab <- read.csv(here::here("analysis","data","derived_data","ITS1_bac_phyla_rel_ab.csv"))

V4_taxa_all_sample <- 
  V4_taxa_ab %>%   # OTU_df_V4_original should be ran in 000-tidy-data
  group_by(label) %>% 
  mutate(total_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  group_by(Phylum, label) %>% 
  mutate(each_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  distinct(label, Phylum, total_phylum_sum, each_phylum_sum) %>% 
  mutate(rel_ab = round(100*each_phylum_sum/total_phylum_sum, 1))

pottery_GJB <-V4_taxa_all_sample %>% filter(!str_detect(label, "pot6")&!str_detect(label, "soil")&!str_detect(label, "con"))
pottery_GS <-V4_taxa_all_sample %>% filter(str_detect(label, "pot6"))
Firmi <- pottery_GJB %>% filter(Phylum == "p__Firmicutes")
Actin <- pottery_GJB %>% filter(Phylum == "p__Actinobacteria")
Deino <- pottery_GJB %>% filter(Phylum == "p__Deinococcus")
Prote <- pottery_GJB %>% filter(Phylum == "p__Proteobacteria")
pot6_Actin <- pottery_GS %>% filter(Phylum == "p__Actinobacteria")
pot6_Crena <- pottery_GS %>% filter(Phylum == "p__Crenarchaeota")

soil <-V4_taxa_all_sample %>% filter(str_detect(label, "soil"))
Firmi_soil <- soil %>% filter(Phylum == "p__Firmicutes"& label%in%c("soil1", "soil2", "soil3"))
Thau_soil <- soil %>% filter(Phylum == "p__Thaumarchaeota"& label%in%c("soil1", "soil2", "soil3"))
Actin_soil4 <- soil %>% filter(str_detect(label, "soil4") & Phylum == "p__Actinobacteria")
Actin_soil5 <- soil %>% filter(str_detect(label, "soil5") & Phylum == "p__Actinobacteria")
Thau_soil4 <- soil %>% filter(str_detect(label, "soil4") & Phylum == "p__Thaumarchaeota")
Thau_soil5 <- soil %>% filter(str_detect(label, "soil5") & Phylum == "p__Thaumarchaeota")

library(corrr)
# function to create a wide dataframe
rel_ab_wider <- function(df){
  df %>% 
  select(label, Phylum, each_phylum_sum) %>% #each_phylum_sum, rel_ab
  pivot_wider(names_from = label, values_from = each_phylum_sum) %>%  
  select(-Phylum) 
}

rel_ab_wider_v4 <- rel_ab_wider(V4_taxa_all_sample)

# compute correlation test
cor_rel_ab_v4 <- rel_ab_wider_v4 %>%  correlate()

# Pearson correlation test 
calc_cor_p_value <- function(vec_a, vec_b){
  cor.test(vec_a, vec_b)$p.value
}
p_value_cor_v4 <- colpair_map(rel_ab_wider_v4, calc_cor_p_value)

# Wilcoxon rank-sum
calc_wil_p_value <- function(vec_a, vec_b){
  wilcox.test(vec_a, vec_b, p.adjust.method = 'fdr')$p.value
}
p_value_wil_v4 <- colpair_map(rel_ab_wider_v4, calc_wil_p_value)

major_taxa <- function(df, vec1){
  df %>% 
  filter(Phylum == vec1) %>% 
  group_by(label) %>% 
  mutate(total_sum = sum(count)) %>% 
  ungroup() %>% 
  group_by(Class, label) %>% 
  mutate(each_sum = sum(count)) %>% 
  ungroup() %>% 
  distinct(label, Class, total_sum, each_sum) %>% 
  mutate(rel_ab = round(100*each_sum/total_sum, 1))
}

V4_Firmi <- major_taxa(V4_taxa_ab, "p__Firmicutes")
V4_Actin <- major_taxa(V4_taxa_ab, "p__Actinobacteria")
V4_Deino <- major_taxa(V4_taxa_ab, "p__Deinococcus")
V4_Prote <- major_taxa(V4_taxa_ab, "p__Proteobacteria")

V6_taxa_all_sample <- 
  V6_taxa_ab %>%  
  group_by(label) %>% 
  mutate(total_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  group_by(Phylum, label) %>% 
  mutate(each_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  distinct(label, Phylum, total_phylum_sum, each_phylum_sum) %>% 
  mutate(rel_ab = round(100*each_phylum_sum/total_phylum_sum, 1))

rel_ab_wider_v6 <- rel_ab_wider(V6_taxa_all_sample)
p_value_wil_v6 <- colpair_map(rel_ab_wider_v6, calc_wil_p_value)
pottery_v6 <-V6_taxa_all_sample %>% filter(str_detect(label, "pot"))
Pates <- pottery_v6 %>% filter(Phylum == "p__Patescibacteria")
V6_Pates <- major_taxa(V6_taxa_ab, "p__Patescibacteria")

ITS1_taxa_all_sample <- 
  ITS1_taxa_ab %>%  
  group_by(label) %>% 
  mutate(total_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  group_by(Phylum, label) %>% 
  mutate(each_phylum_sum = sum(count)) %>% 
  ungroup() %>% 
  distinct(label, Phylum, total_phylum_sum, each_phylum_sum) %>% 
  mutate(rel_ab = round(100*each_phylum_sum/total_phylum_sum, 1))

rel_ab_wider_its1 <- rel_ab_wider(ITS1_taxa_all_sample)
p_value_wil_its1 <- colpair_map(rel_ab_wider_its1, calc_wil_p_value)
Ascom_GJB <- ITS1_taxa_all_sample %>% filter(str_detect(label, "GJB") & Phylum == "p__Ascomycota")
Ascom_GS <- ITS1_taxa_all_sample %>% filter(str_detect(label, "GS") & Phylum == "p__Ascomycota")
Basid_GJB <- ITS1_taxa_all_sample %>% filter(str_detect(label, "GJB") & Phylum == "p__Basidiomycota")
Basid_GS <- ITS1_taxa_all_sample %>% filter(str_detect(label, "GS") & Phylum == "p__Basidiomycota")
```

## Taxonomic composition 
For the V4 region, bacterial composition at the phylum level in pottery interior samples was similar to those of pottery exterior samples and control samples, but slightly different from that of soil samples (@fig-rel-ab: A). For Guijiabao pottery samples, *Firmicutes* (`r min(Firmi$rel_ab)`- `r max(Firmi$rel_ab)`%) was the most abundant phylum, followed by *Actinobacteria* (`r min(Actin$rel_ab)`- `r max(Actin$rel_ab)`%), *Deinococcus* (`r min(Deino$rel_ab)`- `r max(Deino$rel_ab)`%), and *Proteobacteria* (`r min(Prote$rel_ab)`- `r max(Prote$rel_ab)`%). Gaoshan sample is predominated by *Actinobacteria* (`r min(pot6_Actin$rel_ab)`- `r max(pot6_Actin$rel_ab)`%) with the presence of *Crenarchaeota* (`r min(pot6_Crena$rel_ab)`- `r max(pot6_Crena$rel_ab)`%) belonging to archaeal domain that was not found in Guijiabao samples and control samples. Although a different composition of phylum was observed between Guijiabao and Gaoshan samples, the difference is not statistically significant. The soil samples from Guijiabao is predominated by *Firmicutes* (`r min(Firmi_soil$rel_ab)`- `r max(Firmi_soil$rel_ab)`%) and *Thaumarchaeota* (`r min(Thau_soil$rel_ab)`- `r max(Thau_soil$rel_ab)`%) belonging to the archaeal domain, while soil samples from Gaoshan is predominated by *Actinobacteria* (`r Actin_soil4$rel_ab`- `r Actin_soil5$rel_ab`%) with less abundant *Thaumarchaeota* (`r Thau_soil4$rel_ab`- `r Thau_soil5$rel_ab`%). 

```{r abundance-significant}
#| label: abundance-significant
#| echo: false
#| cache: true
library(broom)
library(Hmisc)
# function to read data
shared_rarefy_long <- function(filename, vec){
  read.csv(here::here("analysis","data","derived_data", filename), sep = vec) %>%
    select(Group, starts_with("ASV")) %>%
    pivot_longer(-Group, names_to="otu", values_to="count")
}

shared_V4_rarefy <-shared_rarefy_long("V4_shared_format.1.subsample.60216.csv", ",")
shared_V1_rarefy <-shared_rarefy_long("V1_shared_format.1.subsample.13658.csv", "")
shared_V6_rarefy <-shared_rarefy_long("V6_shared_format.1.subsample.28402.csv", ",")
shared_ITS1_rarefy <-shared_rarefy_long("ITS1_shared_format.1.subsample.387.csv", ",")

# function to tidy taxonomy data
tidy_taxonomy <- function(filename){
  read.table(here::here("analysis","data","raw_data",filename), header = T, sep="\t") %>%
  select(OTU.ID, taxonomy) %>%
  rename(otu = OTU.ID) %>%
  mutate(taxonomy = str_replace_all(taxonomy, "\\(\\d+\\)", ""),
         taxonomy = str_replace(taxonomy, "; c__; o__; f__; g__; s__", ""),
         taxonomy = str_replace(taxonomy, "; o__; f__; g__; s__", ""),
         taxonomy = str_replace(taxonomy, "; f__; g__; s__", ""),
         taxonomy = str_replace(taxonomy, "; g__; s__", ""),
         taxonomy = str_replace(taxonomy, "(; s__)([^(; s__)]*)$", ""),
         taxonomy = str_replace(taxonomy, "(; f__)([^(; f__)]*)$", ""),
         taxonomy = str_replace(taxonomy, "(; g__)([^(; g__)]*)$", ""))
  }

taxonomy_V4 <- tidy_taxonomy("V4_asv_table.txt")
taxonomy_V1 <- tidy_taxonomy("V1_asv_table.txt")
taxonomy_V6 <- tidy_taxonomy("V6_asv_table.txt")
taxonomy_ITS1 <- tidy_taxonomy("ITS1_asv_table.txt")

# function to read metadata for each region
meta_region_extract <- function(filename, vec){
  read_excel(here::here("analysis","data", "raw_data", filename)) %>%
    filter(Region == vec) %>%
    rename(Group = `adpater no`) %>%
    mutate(type= str_extract(Group, "[[:alpha:]]+(?=\\d)"), type = tolower(type)) %>%
    mutate(group = if_else(!is.na(area)&!area%in% c("belly", "outside belly"),
                           paste0(type, "-", area), type),
           pot = case_when(group == "pot-inside" ~ "interior", group == "pot-outside" ~"exterior"),
           potex_soil = case_when(group == "pot-outside" ~ "pot-exterior", type== "soil" ~ "soil"),
           potin_soil = case_when(group == "pot-inside" ~ "pot-interior", type== "soil" ~ "soil"),
           pot_soil = case_when(type== "pot" ~ "pot", type== "soil" ~ "soil"),
           pot_con = case_when(type== "pot" ~ "pot", type== "con" ~ "control"),
           potex_con = case_when(group == "pot-outside" ~ "pot-exterior", type== "con" ~ "control"),
           potin_con = case_when(group == "pot-inside" ~ "pot-interior", type== "con" ~ "control")) %>%
    relocate(Group, type, group, sample, area, site)
}

metadata_V4 <- meta_region_extract("GJB_amp_meta.xlsx", "V4")
metadata_V1 <- meta_region_extract("GJB_amp_meta.xlsx", "V1")
metadata_V6 <- meta_region_extract("GJB_amp_meta.xlsx", "V6")
metadata_ITS1 <- meta_region_extract("GJB_amp_meta.xlsx", "ITS1")

# function to calculate relative abundance for taxa level, not ASV level
composite_region <- function(df1, df2, df3){
  inner_join(df1, df2, by= "otu") %>%
    group_by(Group, taxonomy) %>%
    dplyr::summarize(count = sum(count), .groups="drop") %>%
    group_by(Group) %>%
    mutate(rel_abund = count / sum(count)) %>%
    ungroup() %>%
    select(-count) %>%
    inner_join(., df3, by = "Group") %>%
    filter(!taxonomy == "k__Bacteria")
}

composite_V4 <- composite_region(shared_V4_rarefy, taxonomy_V4, metadata_V4)
composite_V1 <- composite_region(shared_V1_rarefy, taxonomy_V1, metadata_V1)
composite_V6 <- composite_region(shared_V6_rarefy, taxonomy_V6, metadata_V6)
composite_ITS1 <- composite_region(shared_ITS1_rarefy, taxonomy_ITS1, metadata_ITS1)

# function to separate the taxonomy data, also in 000-tidy-data
tax_clean_sep <- function(df){
  df %>%
    separate(taxonomy, into = c("Domain", "Phylum", "Class", "Order", "Family", "Genus", "Species"), sep=";") %>%
    mutate(Phylum = str_trim(Phylum), Class = str_trim(Class), 
           Order = str_trim(Order),
           Family = str_trim(Family), Genus = str_trim(Genus), 
           Species = str_trim(Species)) %>%
    mutate(Domain = ifelse(Domain == "Unassigned", "unclassified", Domain),
           Phylum = ifelse(is.na(Phylum), "unclassified", Phylum),
           Class = ifelse(is.na(Class), "unclassified", Class),
           Order = ifelse(is.na(Order), "unclassified", Order),
           Family = ifelse(is.na(Family), "unclassified", Family),
           Genus = ifelse(is.na(Genus), "unclassified", Genus),
           Species = ifelse(is.na(Species), "unclassified", Species)) %>%
    mutate(Phylum = ifelse(Phylum == "p__WPS_2", "p__Eremiobacterota", Phylum),
           Phylum = ifelse(Phylum == "p__Deinococcus_Thermus","p__Deinococcus", Phylum))
}

composite_V4_sep <- tax_clean_sep(composite_V4)
composite_V6_sep <- tax_clean_sep(composite_V6)
composite_V1_sep <- tax_clean_sep(composite_V1)
composite_ITS1_sep <- tax_clean_sep(composite_ITS1)

# function for adding up relative abundance based on different level
composite_sep_genera <- function(df, col){ # col= Family, Genus etc.
  df %>%
    group_by(Group, {{col}}) %>%
    mutate(total_genera_sum = sum(rel_abund)) %>%
    ungroup() %>%
    distinct(Group, {{col}}, group, total_genera_sum, pot_con, pot,
             potex_con, potin_con, potin_soil, potex_soil, pot_soil)
}

composite_V4_sep_family <- composite_sep_genera(composite_V4_sep, Family)
composite_V4_sep_genus <- composite_sep_genera(composite_V4_sep, Genus)
composite_V4_sep_species <- composite_sep_genera(composite_V4_sep, Species)
composite_V6_sep_family <- composite_sep_genera(composite_V6_sep, Family)
composite_V6_sep_genus <- composite_sep_genera(composite_V6_sep, Genus)
composite_V6_sep_species <- composite_sep_genera(composite_V6_sep, Species)
composite_V1_sep_family <- composite_sep_genera(composite_V1_sep, Family)
composite_V1_sep_genus <- composite_sep_genera(composite_V1_sep, Genus)
composite_V1_sep_species <- composite_sep_genera(composite_V1_sep, Species)
composite_ITS1_sep_family <- composite_sep_genera(composite_ITS1_sep, Family)
composite_ITS1_sep_genus <- composite_sep_genera(composite_ITS1_sep, Genus)
composite_ITS1_sep_species <- composite_sep_genera(composite_ITS1_sep, Species)
```

```{r plots-abundance-significant}
#| label: plots-abundance-significant
#| echo: false
#| cache: true
# get significant genera at family level between pottery and control
sig_family_V4_potex_con <-
  composite_V4_sep_family %>% # replace the region
  nest(data = -Family) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ potex_con, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05)

sig_family_V4_pot <-
  composite_V4_sep_family %>% # replace the region
  nest(data = -Family) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ pot, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05)

sig_genus_V4_pot <-
  composite_V4_sep_genus %>% # replace the region
  nest(data = -Genus) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ pot, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05) %>% 
  filter(!Genus == "unclassified")

V4_sig_family_all <- rbind(sig_family_V4_potex_con, sig_family_V4_pot) %>% slice(1,2,5)

# join the original data of significance test 
composite_V4_genus_for_plot <-
  composite_V4_sep_genus %>%
  inner_join(sig_genus_V4_pot, by = "Genus") %>%
  mutate(Genus = str_remove(Genus, "g__")) %>% 
  rename(Taxon = Genus)

composite_V4_family_for_plot <-
  composite_V4_sep_family %>%
  inner_join(V4_sig_family_all, by = "Family") %>%
  mutate(Family = str_remove(Family, "f__")) %>% 
  rename(Taxon = Family)

# V4 plot
library(ggtext)
composite_V4_family_genus_com_plot <-
  rbind(composite_V4_genus_for_plot, composite_V4_family_for_plot) %>%
  mutate(Taxon = str_replace(Taxon, "(.*)", "*\\1*"),
         Taxon = str_replace(Taxon, "\\*Candidatus_(.*)\\*", "Candidatus<br>*\\1*")) %>%
  mutate(total_genera_sum = 100 * (total_genera_sum + 1/15000), # make it percentage and add a very small number
         pot = factor(group, levels = c("con", "pot-inside", "pot-outside", "soil"))) %>%
  ggplot(aes(x= total_genera_sum, y = Taxon, fill= group)) +
  geom_boxplot(aes(fill = group), color = "gray60", show.legend = TRUE) +
  scale_x_log10() +
  scale_fill_viridis_d(breaks = c("con", "pot-inside", "pot-outside", "soil"),
                       labels = c("control","pot-interior", "pot-exterior", "soil"), direction = -1) +
  scale_y_discrete(limits=rev) +
  labs(x= "Relative abundance (%)", y= NULL) +
  theme_minimal() +
  theme(axis.text.y = element_markdown(size= 12), legend.position = "none")

sig_family_V6_pot <-
  composite_V6_sep_family %>% # replace the region
  nest(data = -Family) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ pot, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05)

sig_family_V6_potin_con <-
  composite_V6_sep_family %>% # replace the region
  nest(data = -Family) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ potin_con, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05)

sig_genus_V6_potin_con <-
  composite_V6_sep_genus %>% # replace the region
  nest(data = -Genus) %>%
  mutate(test = map(.x= data, ~wilcox.test(total_genera_sum~ potin_con, data=.x, exact = FALSE) %>% tidy)) %>% #run broom first
  unnest(test) %>%
  filter(p.value < 0.05)

V6_sig_family_all <- rbind(sig_family_V6_pot, sig_family_V6_potin_con) %>% slice(1:2)

# join the original data of significance test 
composite_V6_genus_for_plot <-
  composite_V6_sep_genus %>%
  inner_join(sig_genus_V6_potin_con, by = "Genus") %>%
  mutate(Genus = str_remove(Genus, "g__")) %>% 
  rename(Taxon = Genus)

composite_V6_family_for_plot <-
  composite_V6_sep_family %>%
  inner_join(V6_sig_family_all, by = "Family") %>%
  mutate(Family = str_remove(Family, "f__")) %>% 
  rename(Taxon = Family)

# V6 plot
composite_V6_family_genus_com_plot <-
  rbind(composite_V6_genus_for_plot, composite_V6_family_for_plot) %>%
  mutate(Taxon = str_replace(Taxon, "(.*)", "*\\1*")) %>%
  mutate(total_genera_sum = 100 * (total_genera_sum + 1/15000), # make it percentage and add a very small number
         pot = factor(group, levels = c("con", "pot-inside", "pot-outside", "soil"))) %>%
  ggplot(aes(x= total_genera_sum, y = Taxon, fill= group)) +
  geom_boxplot(aes(fill = group), color = "gray60", show.legend = TRUE) +
  scale_x_log10() +
  scale_fill_viridis_d(breaks = c("con", "pot-inside", "pot-outside", "soil"),
                       labels = c("control","pot-interior", "pot-exterior", "soil"),
                       direction = -1) +
  scale_y_discrete(limits=rev) +
  labs(x= "Relative abundance (%)", y= NULL) +
  theme_minimal() +
  theme(axis.text.y = element_markdown(size= 12),
        legend.title=element_text(size=12),
        legend.text=element_text(size=12))

library(cowplot)
V4_V6_rel_sig_taxa_combine <-
  plot_grid(composite_V4_family_genus_com_plot, composite_V6_family_genus_com_plot, rel_widths = c(0.8, 1), labels = "AUTO", ncol = 2)

ggsave(here::here("analysis","figures","V4_V6_rel_sig_taxa_combine.png"),
       width = 10, height = 4, dpi = 360, units = "in")
```

The dominant classes include *Bacilli*, *Clostridia*, and *Negativicutes* within *Firmicutes*; *Actinobacteria*, *Rubrobacteria*, *Thermoleophilia* within *Actinobacteria*, *Deinococci* within *Deinococcus*, and *Gammaproteobacteria*, *Alphaproteobacteria*, and *Deltaproteobacteria* within *Proteobacteria*. At the family level, significant differences were observed between pottery exterior samples and control samples, and between pottery exterior and interior samples. *Paenibacillaceae*, *Erysipelotrichaceae*, and *Xiphinematobacteraceae* were more abundant in pottery exterior samples than those in control samples (P <0.05; @fig-rel-ab-significant-plot: A). In addition to *Paenibacillaceae* and *Xiphinematobacteraceae*, *Phycisphaeraceae* was also more abundant in pottery exterior samples than in pottery interior samples (P <0.05). At the genus level, Candidatus *Xiphinematobacter* was more abundant in pottery exterior samples compared to pottery interior samples and control samples (P <0.01). Among the bacterial taxa showing significant differences, only *Paenibacillaceae* in the pottery exterior sample displayed relative abundances exceeding 10% on average, while others were below 1%.

The V6 region has similar bacterial communities at the phylum level as the V4 region, with the exception of *Patescibacteria* (`r min(Pates$rel_ab)`- `r max(Pates$rel_ab)`%) in pottery samples, which was not detected in the reads of the V4 region (@fig-rel-ab: B). The dominant classes within *Patescibacteria* include *Parcubacteria*, *Microgenomatia*, and *Saccharimonadia*. We found a significant difference in the relative abundance of phyla between the pottery interior sample and soil samples from Gaoshan (P < 0.05), but none for Guijiabao samples. At the family level, *Paenibacillaceae* and *67_14* was more abundant in pottery exterior samples than in pottery interior samples (P <0.05; @fig-rel-ab-significant-plot: B). At the genus level, *Sphingomonas* was more abundant in the pottery interior samples than in the control samples (P <0.05). 
The V1 and ITS1 regions have higher proportions of unidentified reads than V4 and V6 regions (@fig-rel-ab: C-D). The bacterial communities at the phylum level for the V1 region are similar to those for the V4 and V6 regions, but there were no significant variations in composition between samples. For the ITS1 region of 18S rRNA, fungal communities at the phylum level were predominated by *Ascomycota* (`r min(Ascom_GJB$rel_ab)`- `r max(Ascom_GJB$rel_ab)`% for Guijiabao samples and `r min(Ascom_GS$rel_ab)`- `r max(Ascom_GS$rel_ab)`% for Gaoshan samples) and *Basidiomycota* (`r min(Basid_GJB$rel_ab)`- `r max(Basid_GJB$rel_ab)`% for Guijiabao samples and `r min(Basid_GS$rel_ab)`- `r max(Basid_GS$rel_ab)`% for Gaoshan samples. 

```{r}
#| label: fig-rel-ab
#| echo: false
#| fig-cap: "A, B, C: The taxonomic composition of bacterial communities in pottery interior, pottery exterior, soil, and control samples based on sequences of the V4, V6, and V1 regions. D: The taxonomic composition of bacterial communities based on sequences of the ITS1 region."
library(here)
knitr::include_graphics(here("analysis", "figures", "all_rel_ab_combine.png"))
```

```{r}
#| label: fig-rel-ab-significant-plot
#| echo: false
#| fig-cap: "A: Bacterial communities at the family and genus level differed significantly in relative abundance between pottery exterior and control samples, and between pottery exterior and interior samples based on sequences of the V4 region. B: Bacterial communities at the family and genus level differed significantly in relative abundance between pottery exterior and interior samples, and between pottery interior and control samples based on sequences of the V6 region."
library(here)
knitr::include_graphics(here("analysis", "figures", "V4_V6_rel_sig_taxa_combine.png"))
```

## Beta diversity analysis
We used principal coordinates analysis (PCoA) based on binary Jaccard and Bray-Curtis dissimilarity to evaluate any structural differences between bacterial and fungal communities in four groups: control, pottery interior, pottery exterior, and soil samples. The bacterial communities identified by the V1, V4, and V6 region sequences showed a significant difference between four sample groups for their presence and abundance according to PERMANOVA tests (P < 0.01), while there were no significant differences in the fungal communities (@fig-V4-V6-PCoA). A further pairwise PERMANOVA tests showed bacterial communities in pottery interior and exterior samples differed significantly from soil samples (P < 0.05) based on the V4 and V6 region sequences, but not on the V1 region sequences. Although there were no significant differences in beta diversity between pottery interior, pottery exterior, and control samples, the Jaccard PCoA plot (the V4 region) indicated a higher intra-variability between pottery samples from Guijiabao and Gaoshan, as well as soil samples from two sites according to the Jaccard PCoA plot (the V6 region) and the Bray-Curtis PCoA plots (the V4 and V6 regions).

```{r}
#| label: fig-V4-V6-PCoA
#| echo: false
#| cache: true
#| fig-cap: "A: PCoA plots of bacterial communities based on the sequences of V4 region. B: PCoA plots of bacterial communities based on the sequences of V6 region. The percentage variability explained by each principal component is indicated on the axes"
knitr::include_graphics(here("analysis", "figures", "V4_V6_combine_PcoA.png"))
```

## Shared and unique microbial communities
Pairwise comparison of taxonomy of each amplicon sequence variant (ASV) among different sample groups was conducted to investigate any unique microbial communities particularly in pottery samples. For the sequence data of 16S rRNA regions, V6 sequences show higher number of unique taxonomy (46, 10%) in pottery interior samples compared to those of V1 sequences (27, 19.3%) and V4 sequences (7, 1.1%), wheres the sequence of 18S rRNA based on ITS1 region presents no specific taxonomy in pottery interior samples (@fig-venn-diagram). Similar patterns were observed for pottery exterior samples, where V6 sequences present a higher number of unique taxonomy (42, 9.1%), followed by V1 sequences (11, 7.9%), V4 sequences (8, 1.3%), and ITS1 sequences (4, 9.8%). A close look at the unique taxa

```{r}
#| label: fig-venn-diagram
#| echo: false
#| cache: true
#| fig-cap: "Venn diagrams based on reads from each regions analyzed in this study showing shared and specific species for bacterial and fungal communities for four sample groups"
knitr::include_graphics(here("analysis", "figures", "all_taxa_venn_diagram.png"))
```

Questions:

1. Abundance and relative abundance? Proportion of the general microbial communities across different types of analysis. Richness and Shanon analysis
2. What should I focus on heatmap? those high frequencies? targets? or based on tree? 
3. How to deal with control sets? Subtract? 
4. What is the correction between V1, V4, V6 of 16S according to abundance and taxonomy?
5. Phylogentic-based analysis, phylogenic tree?
6. the distribution of microbiome across different groups of samples?

ANOVA
normal distribution
standard deviation 
Kristall_willis

# Discussion

There are significant differences in the richness and diversity of bacterial and archaeal communities across the four groups: soil > exterior potsherds > interior potsherds > control
Some microbial groups areÂ significantly more abundantÂ in pottery samples, such as Weissella, Escherichia Shigella, Paenibacillus, and Chryseobacterium, and Penicillium
V4 and V6 regions are more ideal for determining the microorganisms in archaeological materials compared with V1 and ITS1 regions 


# Conclusion

# Acknowledgements

<!-- The following line inserts a page break  -->

\newpage

# References

<!-- The following line ensures the references appear here for the MS Word or HTML output files, rather than right at the end of the document (this will not work for PDF files):  -->

::: {#refs}
:::

\newpage

### Colophon

This report was generated on `r Sys.time()` using the following computational environment and dependencies:

```{r}
#| label: colophon
#| cache: false

# which R packages and versions?
if ("devtools" %in% installed.packages()) devtools::session_info()
```

The current Git commit details are:

```{r}
# what commit is this file at? 
if ("git2r" %in% installed.packages() & git2r::in_repository(path = ".")) git2r::repository(here::here())  
```
